document/js-api/index.bs

```diff
@@ -1329,6 +1383,165 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
 
 </div>
 
+<h3 id="jspi">JavaScript Promise Integration</h3>
+
+Note: The JavaScript Promise Integration API (JSPI) allows WebAssembly functions to suspend and resume their execution -- based on the behavior of JavaScript functions that return Promise objects.
+
+A {{Suspending}} object represents a JavaScript function whose calls via WebAssembly imports should be *suspended* when they return a Promise object.
```

Comment:
It is kind of important that the wrapped entity does not count as a function:
1. It is referenced in 'add-the-imports' as a separate case.
2. Allowing it as a function can cause some confusion: what would happen if you call it (for example, from JavaScript)?
3. Architecturally, we want to use this as a marker to allow the Module instantiation process to process these imports differently.
4. There is further potential for confusion because an import can be re-exported. If it is a function, then is the export also a wrapped Suspending object, or not; and can the result be re-imported into another module while preserving the Suspending 'flag'.